src/makeCalculator.js

```diff
@@ -5,6 +5,35 @@
  */
 function makeCalculator() {
   // write code here
+  let result = 0;
+
+  return {
+    get result() {
+      return result;
+    },
+    add(num) {
+      result += num;
+    },
+    subtract(num) {
+      result -= num;
+    },
+    multiply(num) {
+      result *= num;
+    },
+    divide(num) {
+      result /= num;
+    },
+    reset() {
+      result = 0;
+
+      return this;
+    },
+    operate(callback, num) {
+      callback.call(this, num);
+
+      return this;
```

Comment:
Ensure that the `callback` provided to `operate` is a function before calling it to avoid runtime errors. You can check this with `typeof callback === 'function'`.